cpp/cmake_modules/FindQPL.cmake

```diff
@@ -0,0 +1,66 @@
+# Copyright 2012 Cloudera Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# - Find QPL 
+#
+# Variables used by this module, they can change the default behaviour and need
+# to be set before calling find_package:
+#
+#  QPL_ROOT - When set, this path is inspected instead of standard library
+#                locations as the root of the Thrift installation.
+#                The environment variable THRIFT_HOME overrides this variable.
+#
+# This module defines
+#  QPL_FOUND, whether Thrift is found or not
+#  QPL_COMPILER_FOUND, whether Thrift compiler is found or not
+#
+#  qpl::qpl, a library target to use QPL
+
+if(QPL_FOUND)
+  return()
+endif()
+
+if(QPL_ROOT)
+  find_library(QPL_STATIC_LIB
+               NAMES qpl
+               PATHS ${QPL_ROOT}
+               NO_DEFAULT_PATH
+               PATH_SUFFIXES ${ARROW_LIBRARY_PATH_SUFFIXES})
+  find_path(QPL_INCLUDE_DIRS
+            NAMES qpl.h
+            PATHS ${QPL_ROOT}
+            NO_DEFAULT_PATH
+            PATH_SUFFIXES ${ARROW_INCLUDE_PATH_SUFFIXES})               
+else()
+  find_library(QPL_STATIC_LIB
+               NAMES qpl
+               PATH_SUFFIXES ${ARROW_LIBRARY_PATH_SUFFIXES})
+  find_path(QPL_INCLUDE_DIRS
+            NAMES qpl.h
+            PATH_SUFFIXES ${ARROW_INCLUDE_PATH_SUFFIXES})               
+endif()
+
+if(QPL_STATIC_LIB)
+  set(QPL_FOUND TRUE)
+  add_library(qpl::qpl STATIC IMPORTED)
+  set_target_properties(qpl::qpl
+                        PROPERTIES IMPORTED_LOCATION "${QPL_STATIC_LIB}"
+                                   INTERFACE_INCLUDE_DIRECTORIES "${QPL_INCLUDE_DIRS}")
+else()
+  if(QPL_FIND_REQUIRED)
+    message(FATAL_ERROR "QPL library was required in toolchain and unable to locate")
+  endif()
+  set(QPL_FOUND FALSE)
+endif()
```

Comment:
We can use `find_package_handle_standard_args()`: https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html